src/transport_performance/utils/defence.py

```diff
@@ -90,7 +90,12 @@ def _check_parent_dir_exists(
     return None
 
 
-def _is_expected_filetype(pth, param_nm, check_existing=True, exp_ext=".zip"):
+def _is_expected_filetype(
+    pth: Union[pathlib.Path, str],
+    param_nm: str,
+    check_existing: bool = True,
+    exp_ext: Union[str, list] = ".zip",
+) -> None:
```

Comment:
It is very likely that people pass file extensions to exp_ext without a leading '.' e.g., 'html' rather than '.html'. It may be useful to either detect and adjust this, or to mention is explicitly in the docstring

Reply:
Yeah did think about that. OK, I'll probably handle that case and put something in the docstring, as I'm very likely the one who will misuse it in that way,